### DISCUSSION TITLE
How to hook on RECONFIGURE6?

### DISCUSSION THREAD

**Discussion by tuxmike** [rsmarples/dhcpcd#162] [Q&A] [answered]
Hi,
currently trying to hook a script on new ISP prefix assignment. My router sends a RECONFIGURE6 msg to my host running dhcpd, but somehow, it is not possible to create a dhcpd hook on that?
Currently, i'm using the subsequent BOUND6 hook, but that happens a lot later (~20 min), so i'm loosing time in propagation / connectivity.

syslog:
> Jan  8 05:41:06 hostname dhcpcd[18309]: eth0: unauthenticated RECONFIGURE6 from fe80:XXXX:XXXX...
> [...]
> Jan  8 06:01:01 hostname dhcpcd[18309]: eth0: REPLY6 received from fe80:XXXX:XXXX...
> Jan  8 06:01:01 hostname dhcpcd[18309]: eth0: adding address 2XXX:XXXX:...

Any idea how to hook to the RECONFIGURE6 reason to detect the change earlier?
Thanks a lot!

**Comment by rsmarples** [comment, marked as answer]
RECONFIGURE messages need to be authenticated and yours is not.
A reconfigure key should have been sent in the initial SOLICT and REPLY message sequence.
You would need to capture the full message sequence with tcpdump to analyse this.

**Reply by tuxmike** [reply]
Ok, that means, the RECONFIGURE event is dropped, because it is not authenticated? Thanks! 
So I have to figure out, how to enable authentication, that the client directly requests renew/bind on reconfigure, or is it something which has to be set in DHCP server on router side?

**Reply by rsmarples** [reply]
Actually both are possible :)

With Delayed Key Authentication the server sends a key in the initial SOLICIT/REPLY messages.
You should see this with tcpdump.
The key isn't retransmitted beyond this point, it's only done once so you don't see it in the lease file.

OR you configure the realm and key at *both* ends. This is described in the dhcpcd.conf man page.

**Reply by tuxmike** [reply]
As my router does not seem to support auth, I set noauthrequired in the client, an it is working now, thanks for explaining what happened :star_struck:
